trollflow2/plugins/__init__.py

```diff
@@ -279,10 +279,13 @@ def __setstate__(self, kwargs):
         """Set things running even when loading from YAML."""
         LOG.debug('Starting publisher')
         self.port = kwargs.get('port', 0)
-        self.nameservers = kwargs.get('nameservers', None)
-        self.pub = NoisyPublisher('l2processor', port=self.port,
-                                  nameservers=self.nameservers)
-        self.pub.start()
+        self.nameservers = kwargs.get('nameservers', "")
+        if self.nameservers is None:
+            self.pub = Publisher("tcp://*:" + str(self.port), "l2processor")
+        else:
+            self.pub = NoisyPublisher('l2processor', port=self.port,
+                                      nameservers=self.nameservers)
```

Comment:
I think @adybbroe just stumbled on a problem with nameservers set to None, right?